### DISCUSSION TITLE
Converting .png -> .tga, how to set header's last byte?

### DISCUSSION THREAD

**Discussion by FiftyTifty** [ImageMagick/ImageMagick#2401] [Help] [answered]
This is a very obscure use case, and haven't found anything about how to do this, so forgive me if this is fairly trivial.

I have a bunch of low res .tga files that I've converted into PNG -> Upscaled with ESRGAN -> Flipped upside down. Now I need to convert these modified .png images into .tga. But there is a problem. The software that uses the .tga files needs the 18th byte (the last byte) in the header to be "00", but ImageMagick (and XNView) both force it to be "20".

How do I properly convert these .png files using ImageMagick? If it would help, I can archive and upload the original .tga files, modified .png files, and modified .tga files.

**Comment by snibgo** [comment]
According to https://en.wikipedia.org/wiki/Truevision_TGA, that byte is:
```
Image descriptor (1 byte): bits 3-0 give the alpha channel depth, bits 5-4 give direction
```
IM coders\tga calls that byte "attribute". A decimal value of 20 would be hex #14. IM sets the top bits according to orientation. If you want that to be zero, tell IM to "-auto-orient" before writing the output.

EDIT: corrected "-auto-rotate" to "-auto-orient".

I suppose your image needs to have no transparency to ensure the alpha channel depth is zero.

If that doesn't help, please paste a link to your files.

**Reply by FiftyTifty** [reply]
With the following command: **magick mogrify -format tga -auto-rotate *.png**
The following error is outputted: **mogrify: unrecognized option `-auto-rotate' @ error/mogrify.c/MogrifyImageCommand/4212.**

Here are the source .tga files: https://mega.nz/file/Ckt2QY4Y#5zvgZPL-6XxxfUzsAmdp2CJutBXg3MH4iuce8tovkaM
Here are the .png files to convert: https://mega.nz/file/Cw9wwYDQ#SmsnoOGdaxbHeBLgPv39C6rF7FM064NafvOlc7UvFnQ

With the archives, is the tool egtpacker, and a .txt file. To use it, the filenames must be unchanged, with the following command used: **egtpacker headchild pack m2**. If the .tga files have the correct format, they'll be packed into a .egt file.

As you can see, there are 50 images that need to be converted and packed into an .egt file, so being able to covert many files at once is a necessity, as I'm reverse engineering how the layers are applied.

**Reply by fmw42** [reply]
The command is -auto-orient, not -auto-rotate

**Reply by FiftyTifty** [reply]
Thanks for that. The command now passes, but the header byte is still "20" (in hex, not decimal).

**Comment by snibgo** [comment]
With IM v7.0.8-64, this command ...
```
magick headchild00_1.png x.tga
```
... makes x.tga where the first 18 bytes are:
```
{00}{00}{02}{00}{00}{00}{00}{00}{00}{00}{00}{00}{00}{04}{00}{04}↑{00}
```
Thus, the 18th byte of the TGA is zero, which is what you want.

Once you have solved the problem for a single file, you can repeat for all the files either with "mogrify" or a shell loop.

**Reply by FiftyTifty** [reply]
That is strange, as I did that command with IM v.7.0.10 Q16-HDRI 64, and the .tga file still has that byte as 20. I'll try using the same version as you and will report back.

**Comment by snibgo** [comment, marked as answer]
> IM v.7.0.10 Q16-HDRI 64

Please quote the full version number.

Yesterday, I installed the current IM, v7.0.10-27. That version writes hex #20 to the 18th byte, as you say.

The input file has orientation "undefined", so I would expect the "direction" portion of the TGA byte should be zero.

Why are the two IM versions different? I suspect a bug in recent versions of IM.

**Reply by FiftyTifty** [reply]
Here's the complete version info: v.7.0.10-27 Q16 x64 2020-08-10

Seeing how it works for an older version, yeah it will be a bug.
